actors/evm/src/lib.rs

```diff
@@ -0,0 +1,190 @@
+pub mod interpreter;
+mod state;
+
+use {
+    crate::interpreter::{execute, Bytecode, ExecutionState, StatusCode, System, U256},
+    crate::state::State,
+    bytes::Bytes,
+    fil_actors_runtime::{
+        actor_error, cbor,
+        runtime::{ActorCode, Runtime},
+        ActorDowncast, ActorError,
+    },
+    fvm_ipld_blockstore::Blockstore,
+    fvm_ipld_encoding::tuple::*,
+    fvm_ipld_encoding::RawBytes,
+    fvm_ipld_hamt::Hamt,
+    fvm_shared::error::*,
+    fvm_shared::{MethodNum, METHOD_CONSTRUCTOR},
+    num_derive::FromPrimitive,
+    num_traits::FromPrimitive,
+};
+
+#[cfg(feature = "fil-actor")]
+fil_actors_runtime::wasm_trampoline!(EvmContractActor);
+
+/// Maximum allowed EVM bytecode size.
+/// The contract code size limit is 24kB.
+const MAX_CODE_SIZE: usize = 24 << 10;
+
+#[derive(FromPrimitive)]
+#[repr(u64)]
+pub enum Method {
+    Constructor = METHOD_CONSTRUCTOR,
+    InvokeContract = 2,
+}
+
+pub struct EvmContractActor;
+impl EvmContractActor {
+    pub fn constructor<BS, RT>(rt: &mut RT, params: ConstructorParams) -> Result<(), ActorError>
+    where
+        BS: Blockstore,
+        RT: Runtime<BS>,
+    {
+        rt.validate_immediate_caller_accept_any()?;
+
+        if params.bytecode.len() > MAX_CODE_SIZE {
```

Comment:
I'd actually call this field `init_code` because that's what the Ethereum protocol calls it (it's the constructor code).